tests/standalone_2/io/web_socket_test.dart

```diff
@@ -46,9 +46,11 @@ class SecurityConfiguration {
       ? HttpServer.bindSecure(HOST_NAME, 0, serverContext, backlog: backlog)
       : HttpServer.bind(HOST_NAME, 0, backlog: backlog);
 
-  Future<WebSocket> createClient(int port) =>
-      // TODO(whesse): Add client context argument to WebSocket.connect
-      WebSocket.connect('${secure ? "wss" : "ws"}://$HOST_NAME:$port/');
+  Future<WebSocket> createClient(int port, {String customUserAgent}) =>
+      WebSocket.connect('${secure ? "wss" : "ws"}://$HOST_NAME:$port/',
+          customClient: secure
```

Comment:
Not sure if you are able to run the web_socket_test to troubleshoot the failure on Gerritt, hopefully you are.
Just to confirm - the issue here is that `testShouldSetUserAgent` when run with `SecurityConfiguration` `secure`:`true` fails due to the fact that global `WebSocket.userAgent` is not used in that setting - the connection is made using `customClient` with its `userAgent`.

Reply:
I am not able to actually test it.  However, WebSocket.userAgent comes from a final static, unless something else is setting the user agent, it shouldn't it never get changed?  Would be a lot easier if I could debug it though.

Reply:
You could try following https://github.com/dart-lang/sdk/wiki/Building to get the sdk built locally, test running.
https://github.com/dart-lang/sdk/blob/main/tests/standalone/io/web_socket_test.dart#L550 from the test that [fails](https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket/8836546572039956609/+/u/test_results/new_test_failures__logs_) is setting WebSocket.userAgent directly.

Reply:
Took a while to get it up and running, but it was a simple fix.  Because of how the tests are ran and because WebSocket.userAgent comes from a static final, testShouldSetUserAgent and testStaticClientUserAgentStaysTheSame were setting the userAgent to different values at roughly the same time.  By just making sure that they are the same value, the static final variable doesn't get changed.

Reply:
Thank you, it looks like there are more failures due to use of WebSocket.connect() when `secure` is `true` but without customClient that is needed.

Reply:
I'm going to push a fix to the test to this PR if you don't mind @rnewquist.
It's basically about several places in the test where `WebSocket.connect` is invoked without provided `customClient` which is needed when running in secure configuration

Reply:
@aam that would be perfect, I am a little busy this week, sorry about that.